packages/send/backend/src/trpc/users.ts

```diff
@@ -170,6 +172,81 @@ export const usersRouter = router({
       }
     }),
 
+  // This listener is not exposed to the API, but is used internally to listen for verification events
+  onPassphraseShared: trpc
+    .input(
+      z.object({
+        code: z.string(),
+      })
+    )
+    .subscription(async function* (opts) {
+      if (!opts.ctx?.user?.id) {
+        console.error('Verification can only be done by logged for users');
+        return;
+      }
+      // listen for new events
+      for await (const [data] of on(verificationEmitter, 'shared_passphrase', {
+        // Passing the AbortSignal from the request automatically cancels the event emitter when the request is aborted
+        signal: opts.signal,
+      })) {
+        const post = data;
+        yield post;
+      }
+    }),
+
+  // This listener is not exposed to the API, but is used internally to listen for verification events
+  onVerificationFinished: trpc
+    .input(
+      z.object({
+        code: z.string(),
+      })
+    )
+    .subscription(async function* (opts) {
+      if (!opts.ctx?.user?.id) {
+        console.error('Verification can only be done by logged for users');
+        return;
+      }
+      // listen for new events
+      for await (const [data] of on(
+        verificationEmitter,
+        'verification_complete',
+        {
+          // Passing the AbortSignal from the request automatically cancels the event emitter when the request is aborted
+          signal: opts.signal,
+        }
+      )) {
+        const post = data;
+        yield post;
+      }
+    }),
```

Comment:
All of this is super fancy :exploding_head: 

So the `for await (const [data] of on(... `  looks for a specific event, grabs the data, and yields it to the front-end. 

Do you have trpc configured to use websockets all the time? (Just curious - I just now saw in the docs that it can do that)